Dockerfile.catalog

```diff
@@ -0,0 +1,14 @@
+# The base image is expected to contain
+# /bin/opm (with a serve subcommand) and /bin/grpc_health_probe
+FROM registry.redhat.io/openshift4/ose-operator-registry:v4.14
+
+# Configure the entrypoint and command
+ENTRYPOINT ["/bin/opm"]
+CMD ["serve", "/configs"]
+
+# Copy declarative config root into image at /configs
+ADD catalog /configs
+
```

Comment:
In [ALBO's catalog.Dockerfile](https://github.com/openshift/aws-load-balancer-operator/blob/721cf7c520603efcc9f81ee7e218bc6c66fda575/catalog.Dockerfile), it looks like opm uses a cache (`"--cache-dir=/tmp/cache"`). Is there a reason we'd want the cached version there but not here?

Reply:
Good catch. I don't remember exactly why they are different. Probably I generated both but for ALBO I also looked at how other operators did this. Both approaches work but I think I would align this one with ALBO (and other operators in `openshift` org).